src/app/layout/header/GlobalHeader.tsx

```diff
@@ -8,14 +8,14 @@ import SignupModal from '@/components/domain/auth/signup/SignupModal';
 import PrismLogo from '@/assets/logo/logo-combine.svg';
 
 export default function GlobalHeader() {
-  const openModal = useModalStore((state) => state.openModal);
+  const { openModal, closeModal } = useModalStore();
 
   const handleOpenLoginModal = () => {
     openModal(<LoginModal />);
   };
 
   const handleOpenSignupModal = () => {
-    openModal(<SignupModal />);
+    openModal(<SignupModal onSuccess={closeModal} onClose={closeModal} />);
```

Comment:
이 부분이 다른 코멘트에서 언급드린 부분입니다.
onClose를 통해 두번 닫히는 것 같습니다.

Reply:
<img width="561" alt="image" src="https://github.com/user-attachments/assets/b0b5a1fa-e8af-4086-841f-0b1295791adb">
로컬에서 실행하니 x버튼으로 닫으면 오류가 나네요 수정 부탁드립니다!

```suggestion
    openModal(<SignupModal onSuccess={closeModal}  />);
```

Reply:
제 로컬에서는 저 오류가 발생하지 않고, 두 번 닫히지도 않습니다.. 저 에러 발생하는 파일이 globalHeader인가요?

https://github.com/user-attachments/assets/9b3466d3-f8bc-4c51-875d-3267652f2f79

Reply:
지금은 제가 로컬로 확인을 못하는 상황이라 .. 
근데 논리적으로 GlobalHeader에서 onClose를 넘겨줄 이유는 없는 것 같습니다 (모달을 닫기만 한다면요. 다른 동작이 필요하다면 넘기는게 맞습니다. 다른 코멘트에 남긴 것 처럼 콜백함수 용도로요)

 ModalLayout에서 기본적으로 x를 눌렀을 때 closeModal 커스텀 훅으로 모달을 닫는데,
 GlobalHeader에서도 onClose로 closeModal 를 또 넘기는게 이해가 잘 되지 않습니다

Reply:
아 위에 답글을 단 줄 알았는데 제가 저장 버튼을 안눌렀었네요. 이 부분은 회원가입 성공 시 [회원가입 모달 닫기 + 로그인 모달 열기] 동작을 추가하기 위해 작성한 코드입니다. 로그인 모달을 여는 부분은 로그인 api 부분 작업하면서 같이 처리하려고 했는데 관련 코멘트를 남기지 않았네요. 요구사항 정의서를 보니 회원가입 버튼 클릭 후 로그인 모달이 뜨도록 기획라셔서 해당 동작이 추가되어야 합니다.
이번 브랜치에서 같이 처리할게요.
<img width="908" alt="스크린샷 2024-07-16 오후 4 37 16" src="https://github.com/user-attachments/assets/f04cf0ef-989e-4e56-ae59-53004ea429fd">

Reply:
그럼 결국 closeModal이 아니라 openModal(<Login~ / >) 을 담은 콜백함수를 담아야 하는 것 아닌가요 ??
회원가입 모달은 ModalLayout 내의 closeModal에서 닫힐테니까요

![image](https://github.com/user-attachments/assets/834960f5-1a60-4eb8-8cba-fac57fa3fb62)


위에 댓글로 수정하신다는게 onClose prop으로 closeModal이 아닌 openModal을 실행하는 콜백을 보내도록 수정하신단 말씀이었을까요?

Reply:
✅ 논의한 대로 수정완료 [f817cd9](https://github.com/PRism-Works/PRism-FE/pull/44/commits/f817cd90dd2bcd1f6747411bf4b330d8944b79ed)